### PR TITLE
bpf: Fix issue when running init.sh on aarch64

### LINKED ISSUE #9898: Cilium enable on aarch64
We have ported Cilium master version on Arm server and deployed Nginx to verify and test k8s cluster that composed by three Arm nodes, the status of the pods and the connection between the nodes are normal.
I hope to contribute to cilium support for Arm64, and Related work includes the following aspects:
1.	Fix compiling and running issue on Arm64;
2.	Add arm64 support for building cilium-related images;
3.	Add multi-arch support for cilium-related images;
4.	Add arm64 support for cilium CI/CD.

### PR DESCRIPTION
**Pull request by Jianlin-lv**
Package libc6-dev`s header files in
    /usr/include/aarch64-linux-gnu/
Add the directory path to support aarch64 compiling.

Signed-off-by: Jianlin Lv <Jianlin.Lv@arm.com>

Fixes: #9898

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9899)

### CODE REVIEW

**Review by aanm — APPROVED** [review]

**Review by vadorovsky — APPROVED** [review]

File: bpf/init.sh [hunk 1]
```diff
@@ -56,6 +56,12 @@ rm $RUNDIR/encap.state 2> /dev/null || true
 # This directory was created by the daemon and contains the per container header file
 DIR="$PWD/globals"
 
+if test "$(uname -m)" = "aarch64" ; then
+	INCLUDEDIR="-I. -I$DIR -I$LIB -I$LIB/include -I/usr/include/aarch64-linux-gnu/"
+else
+	INCLUDEDIR="-I. -I$DIR -I$LIB -I$LIB/include"
+fi
```
Comment (borkmann) [inline bpf/init.sh:63]: Is the whole issue the missing gnu stubs compilation error? If so, could we work around it via avoiding stdint.h include in the code? E.g. see:

https://code.forksand.com/oisf/suricata/commit/7906c521cdde5b1d0eb3ce379b8e343c3055653f

Comment (Jianlin-lv) [reply bpf/init.sh:63]: This issue is not about gnu/stubs, but it is very similar to the second issue mentioned in the patch above, the required header files cannot be found;
so this patch adding include path for builder.

Comment (nickb937) [reply bpf/init.sh:63]: The default Clang search paths for AARCH64 are:

```
#include "..." search starts here:
#include <...> search starts here:
 .
 /usr/local/include
 /usr/lib/llvm-7/lib/clang/7.0.0/include
 /usr/include/aarch64-linux-gnu
 /usr/include
```

But when clang is invoked with `-target bpf`, the default search paths become:

```
#include "..." search starts here:
#include <...> search starts here:
 .
 /usr/local/include
 /usr/lib/llvm-7/lib/clang/7.0.0/include
 /usr/include
```

There are no symlinks in /usr/include to point to the `gnu` and `bits` subdirectories in `aarch64-linux-gnu` so the path for non-X86 targets needs to be explicitly set, per the suggested change. 

Full error is:

```
evel=warning msg="+ clang -O2 -g -target bpf -emit-llvm -Wno-address-of-packed-member -Wno-unknown-warning-option -I. -I/run/cilium/state/globals -I/var/lib/cilium/bpf -I/var/lib/cilium/bpf/include -D__NR_CPUS__=4 -DENABLE_ARP_RESPONDER -DHANDLE_NS -DSECLABEL=2 -DBPF_PKT_DIR=1 '-DNODE_MAC={.addr={0x02,0x6e,0x1f,0x0b,0x64,0xbf}}' -DCALLS_MAP=cilium_calls_overlay_2 -c /var/lib/cilium/bpf/bpf_overlay.c -o -" subsys=datapath-loader
level=warning msg="In file included from /var/lib/cilium/bpf/bpf_overlay.c:18:" subsys=datapath-loader
level=warning msg="In file included from /run/cilium/state/globals/node_config.h:1:" subsys=datapath-loader
level=warning msg="In file included from /var/lib/cilium/bpf/lib/utils.h:21:" subsys=datapath-loader
level=warning msg="In file included from /var/lib/cilium/bpf/include/bpf/api.h:11:" subsys=datapath-loader
level=warning msg="In file included from /var/lib/cilium/bpf/include/linux/type_mapper.h:4:" subsys=datapath-loader
level=warning msg="In file included from /usr/lib/llvm-7/lib/clang/7.0.0/include/stdint.h:61:" subsys=datapath-loader
level=warning msg="/usr/include/stdint.h:26:10: fatal error: 'bits/libc-header-start.h' file not found" subsys=datapath-loader
level=warning msg="#include <bits/libc-header-start.h>" subsys=datapath-loader
level=warning msg="         ^~~~~~~~~~~~~~~~~~~~~~~~~~" subsys=datapath-loader
level=warning msg="1 error generated." subsys=datapath-loader
```

Comment (tklauser) [reply bpf/init.sh:63]: I just created #10204 implementing what @borkmann suggested above. This should avoid the need for `stdint.h` and should hopefully simplify porting to aarch64 and other architectures.

**Comment by Jianlin-lv** [comment]
Fixed by https://github.com/cilium/cilium/pull/10204

**Pull request closed without merging**